anomalib/models/cs_flow/torch_model.py

```diff
@@ -0,0 +1,531 @@
+
+            nodes.append(
+                Node(node_to_permute, ParallelPermute, {"seed": coupling_block}, name=f"permute_{coupling_block}")
+            )
+            nodes.append(
+                Node(
+                    [nodes[-1].out0, nodes[-1].out1, nodes[-1].out2],
+                    ParallelGlowCouplingLayer,
+                    {
+                        "clamp": self.clamp,
+                        "subnet_constructor": CrossConvolutions,
+                        "subnet_args": {
+                            "channels_hidden": self.cross_conv_hidden_channels,
+                            "kernel_size": self.kernel_sizes[coupling_block],
+                        },
+                    },
+                    name=f"fc1_{coupling_block}",
+                )
+            )
+
+        nodes.append(OutputNode([nodes[-1].out0], name="output_end0"))
+        nodes.append(OutputNode([nodes[-2].out1], name="output_end1"))
+        nodes.append(OutputNode([nodes[-3].out2], name="output_end2"))
+        return GraphINN(nodes)
+
+    def forward(self, images) -> Tuple[Tensor, Tensor]:
+        """Forward method of the model.
+
+        Args:
+            images (Tensor): Input images.
+
+        Returns:
+            Tuple[Tensor, Tensor]: During training: Tuple containing the z_distribution for three scales and the sum
+                of log determinant of the Jacobian. During evaluation: Tuple containing anomaly maps and anomaly scores
+        """
+        features = self.feature_extractor(images)
+        if self.training:
+            output = self.graph(features)
+        else:
+            z_dist, _ = self.graph(features)  # Ignore Jacobians
+            anomaly_scores = self._get_anomaly_scores(z_dist)
+            anomaly_maps = self._get_anomaly_maps(z_dist)
+            output = anomaly_maps, anomaly_scores
+        return output
+
+    def _get_anomaly_maps(self, z_dists: Tuple[Tensor], mode: AnomalyMapMode = AnomalyMapMode.ALL) -> Tensor:
+        """Get anomaly maps by taking mean of the z-distributions across channels.
+
+        By default it computes anomaly maps for all the scales as it gave better performance on initial tests.
+        Use ``AnomalyMapMode.MAX`` for the largest scale as mentioned in the paper.
+
+        Args:
+            z_dists (Tensor): z-distributions for the three scales.
+            mode (AnomalyMapMode): Anomaly map mode.
+
+        Returns:
+            Tensor: Anomaly maps.
+        """
+        anomaly_map: Tensor
+        if mode == AnomalyMapMode.ALL:
+            anomaly_map = torch.ones(z_dists[0].shape[0], 1, *self.input_dims[1:]).to(z_dists[0].device)
+            for z_dist in z_dists:
+                anomaly_map *= F.interpolate(
+                    (z_dist**2).mean(dim=1, keepdim=True),
+                    size=self.input_dims[1:],
+                    mode="bilinear",
+                    align_corners=False,
+                )
```

Comment:
It would improve readability if you take the mean computation outside of the interpolate function, so something like:
```suggestion
                mean_z = (z_dist**2).mean(dim=1, keepdim=True)
                anomaly_map *= F.interpolate(
                    mean_z,
                    size=self.input_dims[1:],
                    mode="bilinear",
                    align_corners=False,
                )
```